main.go

```diff
@@ -55,11 +58,20 @@ func main() {
 
 	castaiClient := castai.NewClient(log, castai.NewDefaultRestyClient(), castai.NewDefaultDeltaHTTPClient())
 
-	castailog.SetupLogExporter(remoteLogger, localLog, castaiClient, &loggingConfig)
+	waitForRegistrationCh := make(chan struct{})
+	var registrationChClose sync.Once
+	registrationChCloseBody := func() {
+		close(waitForRegistrationCh)
+	}
+	defer func() {
+		registrationChClose.Do(registrationChCloseBody)
+	}()
```

Comment:
this is too cryptic for "main". see my other comment for moving this responsibility to `castaiClient`